src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -36,13 +38,31 @@ public UUID send(LetterRequest letter, String serviceName) {
 
         final String messageId = generateChecksum(letter);
 
-        log.info("Generated message: id = {} for letter with print queue id = {}", messageId, letter.type);
+        log.info("Generated message: id = {}", messageId);
 
-        byte[] pdf = pdfCreator.create(letter);
-        Letter dbLetter = new Letter(messageId, serviceName, null, letter.type, pdf);
+        Optional<Letter> result = letterRepository.findByMessageIdOrderByCreatedAtDesc(messageId);
 
-        letterRepository.save(dbLetter);
-        return dbLetter.getId();
+        return result.filter(l -> l.getState().equals(LetterState.Created))
+            .map(l -> {
+                UUID id = l.getId();
+                log.info("Same message found already created. Returning {} letter", id);
+                return id;
+            })
+            .orElse(getNewLetterId(letter, messageId, serviceName));
+    }
+
+    private UUID getNewLetterId(LetterRequest letterRequest, String messageId, String serviceName) {
+        byte[] pdf = pdfCreator.create(letterRequest);
+
+        Letter letter = new Letter(messageId, serviceName, null, letterRequest.type, pdf);
+
+        letterRepository.save(letter);
+
+        UUID letterId = letter.getId();
```

Comment:
you might like to combine this with the preceding line: UUID letterId = letterRepository.save(letter).getId();